### ISSUE TITLE
[Tiq Mini M5] Status bar and text context menu highlighting

### ISSUE THREAD

**Issue by alexknop** [sspanak/tt9#478]
Hello,

Strange issue I am seeing when using TT9 is items become highlighted when they shouldn't be. Either the bottom bar is highlighted or pop-up items are highlighted.
![Screenshot_20240326-110929](https://github.com/sspanak/tt9/assets/11878115/0e780f91-51af-4f8b-a505-4ec8ae173e34)
![Screenshot_20240326-110943](https://github.com/sspanak/tt9/assets/11878115/1d445e33-6e67-469a-9ae7-a80ddbc86fe1)

I am curious if you know where this issue may lie. By itself (without TT9 installed) the phone works with the keypad's navigation arrows and select button and it only has Google Keyboard (qwerty on the touchscreen) as its keyboard. Perhaps there are two IMEs interfering?

**Comment by sspanak** [comment]
You can't have two keyboards active. There may be a hotkey manager running in the background, like on Qin F21 Pro, but not two keyboards.

When does highlighting occur? Is it after you send a message? Does it happen in all apps?

**Comment by alexknop** [comment]
right, I am not suspecting 2 keyboards, but what could be managing the navigation keys functions and maybe that is conflicting with TT9
I am only noticing it in the messaging app. It seems to be random. Sometimes I'll select the text box to type and the bottom status bar will already be highlighted like in the photo.

**Comment by alexknop** [comment]
Nevermind, I am also seeing it when using the browser application. So it is not specific to just the messaging app.

**Comment by sspanak** [comment]
> right, I am not suspecting 2 keyboards, but what could be managing the navigation keys functions and maybe that is conflicting with TT9

Well, after discovering that the volume buttons on Qin F21 Pro do not work properly when there is no status icon, I really don't know what could be conflicting anymore...

Anyway, the comments in #300 and #393 suggest that in some cases, when key presses must be ignored, they are not ignored correctly. I am not sure what is the correct way (I think it is not the same on all phones), but you could try out an experimental option: go to Settings, scroll down to the version number and click it. Then look for "Keypad Handling Mode" and change it to something different than "normal". This will cut all key processing within Traditional T9 and return the control to Android. The downside is, you cannot type anything this way, you can just navigate around. But for the purpose of testing, it shouldn't matter. If the problem occurs often enough, just open and close the messaging app a couple of times, and see if it stops getting highlighted.

Another thing is, I have tested on Android up to 11 only. The APK isn't even built for the latest Android API, so there may be compatibility issues, indeed. I do intend to upgrade and at least test in the emulator, but I was thinking of doing some feature requests that have been waiting for a year now.

Finally, have you used another keyboard long enough to verify this isn't a problem with the phone itself?

**Comment by alexknop** [comment]
I tried changing that setting and the bottom bar was still highlighted when I went back to the messaging app.
I think it could be an Android 13 thing. This is the first time I've noticed it and it's with an Android 13. We can park this for now until we can get another Android 13 device to test it on.

**Comment by wspdyb** [comment]
ה Hi, thanks for the keyboard, it's very good.  This matter is related to setting focus in the layout. If you tell me which layout this line is set to, maybe I can fix it.

**Comment by wspdyb** [comment]
Note that this is 2 separate lines, 1 the list of letters, 2 input mode. You have to turn off the focus there

**Comment by sspanak** [comment]
I never set the focus to any element manually. If you want to give it a try, please go ahead.

There are several layout files, corresponding to each MainView size:
- `res/layout/main_small.xml` is the one from the screenshots here and the videos in #640.
- `res/layout/main_numpad.xml` is the full on-screen keypad, used on touchscreen phones. I am not sure if it needs fixing.

These two contain the Suggestion Bar and the Status Bar. They are overlaid on top of each other. The Status Bar is the one that displays the typing mode and the language. The Suggestion Bar displays the word list when you start to type. 

Suggestion Bar files:
- layout: `res/layout/suggestion_list.xml` (when MainView is small) or `res/layout/suggestion_list_numpad.xml` (when MainView is full numpad)
- Java: `java/io/github/sspanak/tt9/ui/tray/SuggestionsBar.java` and `java/io/github/sspanak/tt9/ui/tray/SuggestionsAdapter.java`

Status Bar files:
- layout: a TextView inside the respective `main_small.xml` or `main_numpad.xml`
- Java: `java/io/github/sspanak/tt9/ui/tray/StatusBar.java`

Let me know, in case you have any further questions.

**Comment by sspanak** [comment]
There seems to be no interest in this problem or it is not happening anymore. I am closing this issue for now.
